src/System.Net.Sockets/src/System/Net/Sockets/TCPClient.cs

```diff
@@ -119,6 +120,24 @@ public Socket Client
             }
         }
 
+        internal Socket NonDisposedClient
+        {
+            get
+            {
+                var socket = Volatile.Read(ref _clientSocket);
+
+                if (socket == null)
+                {
+                    if (Volatile.Read(ref _cleanedUp) == 1)
```

Comment:
Why not just make the field volatile instead?

Reply:
I'll change it to `volatile`, it needs to be an `int` because of `Interlocked.CompareExchange`